### PR TITLE
Only clear out publication statuses that don't match the document's current state

### PR DESCRIPTION
**Pull request by joncarlson**
**Background**

A "Draft" document is published from mEditor. The "CMR Subscriber" picks it up and sends it on to CMR_UAT.

After sending it to CMR_UAT, the "CMR Subscriber" sends an acknowledgement back to mEditor.

Before mEditor stores this acknowledgement in the database, mEditor clears out any existing publication statuses, from the db, for the given subscriber (called "target" in the NATS messages, i.e. `cmr` for CMR subscriber)

**The problem**

If the subscriber pushes to **multiple** CMR providers, mEditor will only store **one** of the acknowledgements that comes back.

**The solution**

Don't clear out any publication statuses matching the same state as the document's state.

In other words, if a document is in "Published" state, clear out any publication statuses for **other** states (Draft, Under Review, etc.) but leave publication statuses in the "Published" state

### CODE REVIEW

File: packages/legacy-api/controllers/Meditor.js [hunk 1]
```diff
@@ -55,6 +55,7 @@ async function handlePublicationAcknowledgements(message) {
                 [acknowledgement.statusCode == 200 ? 'publishedOn' : 'failedOn']:
                     Date.now(),
             }),
+            ...(acknowledgement.state && { state: acknowledgement.state }),
```
Comment (joncarlson) [inline packages/legacy-api/controllers/Meditor.js:58]: Store the document state alongside any acknowledgements in the database

File: packages/legacy-api/controllers/Meditor.js [hunk 2]
```diff
@@ -68,6 +69,9 @@ async function handlePublicationAcknowledgements(message) {
                 $pull: {
                     'x-meditor.publishedTo': {
                         target: acknowledgement.target,
+                        // if document is in "Published" state, this would clear out any publication statuses for **other** states
+                        // i.e. any publication statuses that are marked as "Draft" would be cleared out
+                        state: { $ne: acknowledgement.state },
```
Comment (joncarlson) [inline packages/legacy-api/controllers/Meditor.js:74]: Only clear out publication statuses that don't match the acknowledgements state.

If an acknowledgement doesn't have a state, it would also be cleared out

**Review by benforshey — APPROVED** [review]

**Review by KrupaRami — APPROVED** [review]
